edx_proctoring/models.py

```diff
@@ -289,6 +293,24 @@ def get_onboarding_attempts_by_course_id(self, course_id, users=None):
             queryset = queryset.filter(user__in=users)
         return queryset
 
+    # pylint: disable=invalid-name
+    def get_last_verified_proctored_practice_attempt(self, user_id, proctoring_backend):
+        """
+        Returns the user's last verified proctored practice attempt for a specific backend,
+        if it exists. This only considers attempts within the last two years, as attempts
+        before this point are considered expired.
+
+        Parameters:
+            * user_id: ID of the user
+            * proctoring_backend: The name of the proctoring backend
+        """
+        earliest_allowed_date = datetime.now(pytz.UTC) - timedelta(days=730)
```

Comment:
As a follow up PR, I'd prefer to have this number `730` to be in a edx-platform setting specific to Proctoring.